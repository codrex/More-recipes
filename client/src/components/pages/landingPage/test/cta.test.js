import React from 'react';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from '../../../../store/configStore';
import  Cta  from '../cta/cta';

const props = {
  push: jest.fn(),
  signin: jest.fn(),
  signup: jest.fn(),
};

describe('Cta form component ', () => {
  test('expect to match snapshot when signin button is clicked', () => {
    const component = mount(<Cta {...props} />);
    const tree = toJson(component);
    component.find('#login').simulate('click')
    expect(tree).toMatchSnapshot();
    expect(tree).toBeInstanceOf(Object);
  });
  test('expect to match snapshot when signup button is clicked', () => {
    const component = mount(<Cta {...props} />);
    const tree = toJson(component);
    component.find('#signup').simulate('click')
    expect(tree).toMatchSnapshot();
    expect(tree).toBeInstanceOf(Object);
  });
});
