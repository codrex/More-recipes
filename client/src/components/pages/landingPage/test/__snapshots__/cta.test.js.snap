// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cta form component  expect to match snapshot when signin button is clicked 1`] = `
<Cta
  push={[Function]}
  signin={[Function]}
  signup={[Function]}
>
  <div
    className="cta "
  >
    <h1
      className=" text-uppercase display-4 col-xs-11 col-sm-10 col-md-8 col-lg-9"
    >
      ​share​ and view ​awesome​ ​and​ ​exciting​ ​​recipes
    </h1>
    <div
      className="cta-btn-wrapper"
    >
      <Button
        className="btn-primary btn-lg login"
        handleClick={[Function]}
        id="login"
        text="Login"
      >
        <button
          className="btn text-capitalize btn-primary btn-lg login"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
      </Button>
      <Button
        className=" btn-secondary btn-lg reg text-white"
        handleClick={[Function]}
        id="signup"
        text="Create an account"
      >
        <button
          className="btn text-capitalize  btn-secondary btn-lg reg text-white"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Create an account
        </button>
      </Button>
    </div>
  </div>
</Cta>
`;

exports[`Cta form component  expect to match snapshot when signup button is clicked 1`] = `
<Cta
  push={[Function]}
  signin={[Function]}
  signup={[Function]}
>
  <div
    className="cta "
  >
    <h1
      className=" text-uppercase display-4 col-xs-11 col-sm-10 col-md-8 col-lg-9"
    >
      ​share​ and view ​awesome​ ​and​ ​exciting​ ​​recipes
    </h1>
    <div
      className="cta-btn-wrapper"
    >
      <Button
        className="btn-primary btn-lg login"
        handleClick={[Function]}
        id="login"
        text="Login"
      >
        <button
          className="btn text-capitalize btn-primary btn-lg login"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
      </Button>
      <Button
        className=" btn-secondary btn-lg reg text-white"
        handleClick={[Function]}
        id="signup"
        text="Create an account"
      >
        <button
          className="btn text-capitalize  btn-secondary btn-lg reg text-white"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Create an account
        </button>
      </Button>
    </div>
  </div>
</Cta>
`;
